File: 1.4/reverse_an_integer/reverse.py
def reverse_integer(x: int) -> int:
    rev = 0
    sign = -1 if x < 0 else 1
    x = abs(x)

    while x != 0:
        rev = rev * 10 + x % 10
        x //= 10

    rev *= sign

    # Check 32-bit overflow
    if rev < -2**31 or rev > 2**31 - 1:
        return 0

    return rev

if __name__ == "__main__":
    num = int(input("Enter an integer: "))
    print("Reversed integer:", reverse_integer(num))

File: 1.4/reverse_an_integer/reverse.cpp
#include <iostream>
#include <climits> 
using namespace std;

int reverseInteger(int x) {
    long long rev = 0;

    while (x != 0) {
        int digit = x % 10;
        rev = rev * 10 + digit;
        x /= 10;
    }

    // Check 32-bit overflow
    if (rev < INT_MIN || rev > INT_MAX)
        return 0;

    return (int)rev;
}

int main() {
    int number;
    cout << "Enter an integer: ";
    cin >> number;

    int result = reverseInteger(number);
    cout << "Reversed integer: " << result << endl;

    return 0;
}